packages/rocketchat-ui-master/server/inject.js

```diff
@@ -1,14 +1,68 @@
 /* globals Inject */
 
 Inject.rawBody('page-loading', `
-	<div id="initial-page-loading" class="page-loading">
-		<div class="loading">
-			<div class="bounce1"></div>
-			<div class="bounce2"></div>
-			<div class="bounce3"></div>
-		</div>
-	</div>`
-);
+<style>
+.loading {
+	top: 0;
+	right: 0;
+	bottom: 0;
+	left: 0;
+	display: flex;
+	align-items: center;
+	position: absolute;
+	justify-content: center;
+	text-align: center;
+}
+.loading > div {
+	width: 10px;
+	height: 10px;
+	margin: 2px;
+	border-radius: 100%;
+	display: inline-block;
+	-webkit-animation: loading-bouncedelay 1.4s infinite ease-in-out both;
+	animation: loading-bouncedelay 1.4s infinite ease-in-out both;
+}
+.loading .bounce1 {
+	-webkit-animation-delay: -0.32s;
+	animation-delay: -0.32s;
+}
+.loading .bounce2 {
+	-webkit-animation-delay: -0.16s;
+	animation-delay: -0.16s;
+}
+@-webkit-keyframes loading-bouncedelay {
+	0%, 80%, 100% { -webkit-transform: scale(0) }
+	40% { -webkit-transform: scale(1.0) }
+}
+@keyframes loading-bouncedelay {
+	0%, 80%, 100% { transform: scale(0); }
+	40% { transform: scale(1.0); }
+}
+</style>
+<div id="initial-page-loading" class="page-loading">
+	<div class="loading">
+		<div class="bounce1"></div>
+		<div class="bounce2"></div>
+		<div class="bounce3"></div>
+	</div>
+</div>`);
+
+
+RocketChat.settings.get('theme-color-primary-background-color', function(key, value) {
```

Comment:
Will this introduce delay in showing background color by waiting until it calls?   If so maybe a define the default above.. then Inject new style when / if it gets a setting?